### ISSUE TITLE
Ray rllib DQN training too slow

### ISSUE THREAD

**Issue by AhmetFurkanDEMIR** [ray-project/ray#15836]
Hi, I'm trying to run the DQN algorithm on the [gym_super_mario_bros](https://pypi.org/project/gym-super-mario-bros/) environment, I use ray rllib for this. However, even though I have trained for hours, the rewards that come are very low, when I do tests on the environment, Mario can barely come to the half of the first map. I tried many models and reward functions, but I still couldn't solve the problem, I think the problem is with the hyperparameters, can you help me?.

Reward class↓:
```python
class CustomReward(gym.Wrapper):
    def __init__(self, env):
        super(CustomReward, self).__init__(env)
        self._current_score = 0

    def step(self, action):
        state, reward, done, info = self.env.step(action)

        if reward > 0:
            reward *= 8

        else:

            reward -= 1
            reward *= 2

        reward += (info["score"] - self._current_score)/5.
        self._current_score = info["score"]

        if info["flag_get"]:
            reward += 500

        return state, reward, done, info
```

Model↓:
```python
from ray.rllib.agents.dqn.distributional_q_tf_model import DistributionalQTFModel
from ray.rllib.utils.framework import try_import_tf

tf1, tf, tfv = try_import_tf()

class MyKerasQModel(DistributionalQTFModel):
    """Custom model for DQN."""

    def __init__(self, obs_space, action_space, num_outputs, model_config,
                 name, **kw):


        super(MyKerasQModel, self).__init__(
            obs_space, action_space, num_outputs, model_config, name, **kw)

        self.inputs = tf.keras.layers.Input(shape=obs_space.shape, name="observations")

        self.layer_cnn_1 = tf.keras.layers.Conv2D(32, (3, 3), activation='relu', strides=(2, 2))(self.inputs)
        self.layer_cnn_2 = tf.keras.layers.Conv2D(64, (3, 3), activation='relu')(self.layer_cnn_1)
        self.layer_cnn_3 = tf.keras.layers.Conv2D(128, (3, 3), activation='relu', strides=(2, 2))(self.layer_cnn_2)
        self.layer_cnn_4 = tf.keras.layers.Conv2D(64, (3, 3), activation='relu')(self.layer_cnn_3)

        self.layer_lstm_cell = tf.keras.layers.ConvLSTM2D(32, kernel_size=3, activation='relu')(tf.expand_dims(self.layer_cnn_4, axis=1))

        self.layer_flatten = tf.keras.layers.Flatten()(self.layer_lstm_cell)
        self.layer_dense1 = tf.keras.layers.Dense(512, activation='relu')(self.layer_flatten)
        self.layer_dense2 = tf.keras.layers.Dense(num_outputs, activation='softmax')(self.layer_dense1)
        self.base_model = tf.keras.Model(self.inputs, self.layer_dense2)

    def forward(self, input_dict, state, seq_lens):
        model_out = self.base_model(input_dict["obs"])
        return model_out, state

    def metrics(self):
        return {"foo": tf.constant(42.0)}
```

Config and DQN↓:
```python
ModelCatalog.register_custom_model('jointNetwork', MyKerasQModel)
#env.action_space.n

config = {
        'env': 'super_mario_bros',
        'framework': 'tf',
        "train_batch_size": 512,
        'model':{
          'fcnet_hiddens': [7],
          "custom_model": "jointNetwork",
          "custom_model_config": {}
        }
}

register_env('super_mario_bros', env_creator_lambda)

trainer = DQNTrainer(config=config)
```

**Comment by nicolaspi** [comment]
I am not sure but that might be explained by the fact that Rllib enforces eager execution when tensorflow is used: https://github.com/ray-project/ray/blob/2303851c3c234a261e8791b778e59e71f8235260/rllib/agents/trainer.py#L666
